MLAPI/Core/NetworkingManager.cs

```diff
@@ -604,6 +650,57 @@ public SocketTasks StartHost(Vector3? position = null, Quaternion? rotation = nu
             return tasks;
         }
 
+        /// <summary>
+        /// Become new Host
+        /// </summary>
+        internal SocketTasks BecomeHost()
+        {
+            if (!IsClient)
+            {
+                if (LogHelper.CurrentLogLevel <= LogLevel.Normal)
+                    LogHelper.LogWarning("Cannot become host when we weren't connected");
+                return SocketTask.Fault.AsTasks();
+            }
+
+            // Observers
+            for (int i = 0; i < SpawnManager.SpawnedObjectsList.Count; i++)
+            {
+                for (int j = 0; j < ConnectedClientsList.Count; j++)
+                {
+                    ulong clientId = ConnectedClientsList[j].ClientId;
+
+                    if (clientId == ServerClientId || SpawnManager.SpawnedObjectsList[i].CheckObjectVisibility == null || SpawnManager.SpawnedObjectsList[i].CheckObjectVisibility(clientId))
+                    {
+                        SpawnManager.SpawnedObjectsList[i].observers.Add(clientId);
+                    }
+                }
+            }
+
```

Comment:
This will cause some trouble if:

We migrate all the old clientIds. Then a new client connects, it will reclaim one of the IDs. Instead, each object should In my opinion be unspawned, and a property of "previousClientId" should be set. Then when a new client connects and says in it's handshake "hey I'm a returning client. I previously had X clientId", we will simply spawn the objects back and set the new clientIds.